cub/cub/device/dispatch/kernels/transform.cuh

```diff
@@ -339,10 +337,19 @@ _CCCL_DEVICE void memcpy_async_aligned(void* dst, const void* src, unsigned int
   for (unsigned int offset = threadIdx.x * ldgsts_size_and_align; offset < bytes_to_copy;
        offset += BlockThreads * ldgsts_size_and_align)
   {
-    __pipeline_memcpy_async(
-      static_cast<char*>(dst) + offset, static_cast<const char*>(src) + offset, ldgsts_size_and_align);
+    asm volatile(
+      "cp.async.cg.shared.global [%0], [%1], %2, %3;"
+      :
+      : "r"(static_cast<uint32_t>(__cvta_generic_to_shared(static_cast<char*>(dst) + offset))),
+        "l"(static_cast<const char*>(src) + offset),
```

Comment:
Actually, this is not correct, this is internally marked as `_CUDA_PIPELINE_ASM_PTR_CONSTRAINT`, which conditionally is `r` or `l`

Reply:
Looking at the definition of the macro, it seems to me that it is trying to support 32-bit compilation, which is no longer supported AFAIU.